bundles/org.openhab.binding.qbus/src/main/java/org/openhab/binding/qbus/internal/QbusBridgeHandler.java

```diff
@@ -0,0 +1,313 @@
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, message);
+    }
+
+    /**
+     * Put bridge online when error in communication resolved.
+     */
+    public void bridgeOnline() {
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    /**
+     * Schedule future communication refresh.
+     *
+     * @param interval_config Time before refresh in minutes.
+     */
+
+    private void setupRefreshTimer(int refreshInterval) {
+        ScheduledFuture<?> timer = refreshTimer;
+        if (timer != null) {
+            timer.cancel(true);
+            refreshTimer = null;
+        }
+
+        if (refreshInterval == 0) {
+            return;
+        }
+
+        // This timer will check connection with server and client periodically
+        refreshTimer = scheduler.scheduleWithFixedDelay(() -> {
+            QbusCommunication comm = getCommunication();
+
+            if (comm != null) {
+                if (!comm.communicationActive()) {
+                    // Disconnected from Qbus Server, try to reconnect
+                    comm.restartCommunication();
+                    if (!comm.communicationActive()) {
+                        bridgeOffline("No connection with Qbus Server");
+                        return;
+                    }
+                } else {
+                    // Controller disconnected from Qbus client, try to reconnect controller
+                    if (!comm.clientConnected()) {
+                        comm.restartCommunication();
+                        if (!comm.clientConnected()) {
+                            bridgeOffline("No connection with Qbus Client");
+                            return;
+                        }
+                    }
+                }
+            }
+            updateStatus(ThingStatus.ONLINE);
+
+        }, refreshInterval, refreshInterval, TimeUnit.MINUTES);
+    }
+
+    /**
+     * Disposes the Bridge and stops communication with the Qbus server
+     */
+    @Override
+    public void dispose() {
+        ScheduledFuture<?> timer = refreshTimer;
+        if (timer != null) {
+            timer.cancel(true);
+        }
+
+        refreshTimer = null;
+
+        QbusCommunication comm = getCommunication();
+        if (comm != null) {
+            comm.stopCommunication();
+        }
+
+        comm = null;
+    }
+
+    /**
+     * Sets the configuration parameters
+     */
+    protected void readConfig() {
+        this.config = getConfig().as(QbusConfiguration.class);
+    }
+
+    /**
+     * Get the Qbus communication object.
+     *
+     * @return Qbus communication object
+     */
+    public @Nullable QbusCommunication getCommunication() {
+        return this.qbusComm;
+    }
+
+    /**
+     * Get the ip address of the Qbus server.
+     *
+     * @return the ip address
+     */
+    public @Nullable String getAddress() {
+        if (this.config != null) {
+            return this.config.addr;
```

Comment:
The field can be null at this point, because it isn't inittialized in the constructor. So, you need to check if it is not null before accessing it: if (config != null). But this would result in a compiler warning, as it is a field and could theoretically be modified by another thread exactly between the null check and the access in the next line. To mitigate this, you can store the field to a local variable before checking it:
```
Future<?> localPollingJob = pollingJob;
if (localPollingJob != null) {
    localPollingJob.cancel(true);
}
```

There are some more compiler warnings, which can be seen when calling `mvn clean install`.